lib/Runtime/Library/ArrayBuffer.cpp

```diff
@@ -696,6 +701,13 @@ namespace Js
         Recycler* recycler = type->GetScriptContext()->GetRecycler();
         JavascriptArrayBuffer* result = RecyclerNewFinalized(recycler, JavascriptArrayBuffer, buffer, length, type);
         Assert(result);
+
+        // we take the ownership of the buffer and will have to free it so charge it to our quota.
+        if (!recycler->RequestExternalMemoryAllocation(length))
```

Comment:
why not do this in ArrayBuffer's constructor?

Reply:
I was not sure about getting access to recycler while `this` is still not constructed. And was not comfortable with throwing exceptions from a constructor too.

I plan to revisit the whole deal with allocation of external memory and allocation policy. This code may change anyways.
For now we just want to fix what we have.

---
In reply to: [186512931](https://github.com/Microsoft/ChakraCore/pull/5099#discussion_r186512931) [](ancestors = 186512931)

Reply:
Added a tracking bug to get to this later - [https://github.com/Microsoft/ChakraCore/issues/5114](https://github.com/Microsoft/ChakraCore/issues/5114)


---
In reply to: [186617696](https://github.com/Microsoft/ChakraCore/pull/5099#discussion_r186617696) [](ancestors = 186617696,186512931)